airflow/serialization/serialized_objects.py

```diff
@@ -789,6 +825,27 @@ def from_dict(cls, serialized_obj: dict) -> 'SerializedDAG':
             raise ValueError(f"Unsure how to deserialize version {ver!r}")
         return cls.deserialize_dag(serialized_obj['dag'])
 
+    @classmethod
+    def _serialize_dag_params(cls, dag_params: Dict[str, Param]):
+        """ """
+        serialized_params = {}
+        for k, v in dag_params.items():
+            serialized_params[k] = v.__dict__
+            serialized_params[k].update({'__type': f'{v.__class__.__module__}.{v.__class__.__name__}'})
+
+        return serialized_params
+
+    @classmethod
+    def _deserialize_dag_params(cls, encoded_dag_params: Dict) -> Dict[str, Param]:
+        """ """
+        op_params = {}
+        for k, v in encoded_dag_params.items():
+            param_class = import_string(v['__type'])
```

Comment:
This is potentially risky (to import any string, and then instantiate a object of that class) -- everywhere else in Airflow we require classes that we deserilze to be registered in some way to prevent against what I've been calling "object inflation attacks". (The canonical example is from and old Ruby on Rails bug https://codeclimate.com/blog/rails-remote-code-execution-vulnerability-explained/ )

Right now this input is just about trusted, but I've been operating on the principle that it isn't and shouldn't be, as either we'll use this elsewhere in Airflow, or open it up to API submission.

Not sure 100% what the right answer is here, but right now this is against what we've done elsewhere in Airflow.

Options I can think of:

1. Don't allow this to be pluggable -- so this is _only_ allowed to be an `airflow.models.param.Param`
2. Have some kind of registration like we have with OperatorExtraLinks.

Reply:
We'll go ahead with option 1 for now and keep a TODO item for the 2nd one.

Reply:
Done.. will add unit tests as well around this.